File: code/jvm/src/main/kotlin/pt/isel/leic/ptgest/domain/workout/model/WorkoutDetails.kt
package pt.isel.leic.ptgest.domain.workout.model

import pt.isel.leic.ptgest.domain.workout.MuscleGroup

data class WorkoutDetails(
    val id: Int,
    val name: String,
    val description: String?,
    val muscleGroup: List<MuscleGroup>,
    val sets: List<WorkoutSetDetails>
) {
    constructor(workout: Workout, sets: List<WorkoutSetDetails>) :
        this(workout.id, workout.name, workout.description, workout.muscleGroup, sets)
}

File: code/js/src/views/user/TrainerViews/models/workouts/WorkoutDetails.ts
import SetDetails from "../sets/SetDetails.ts";

class WorkoutDetails {
    id: number
    name: string
    description: string | null
    muscleGroup: string[]
    sets: SetDetails[]

    constructor(id: number, name: string, description: string | null, muscleGroup: string[], sets: SetDetails[]) {
        this.id = id;
        this.name = name;
        this.description = description;
        this.muscleGroup = muscleGroup;
        this.sets = sets;
    }
}
export default WorkoutDetails;
